lib/spack/spack/test/cmd/fetch.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from six import StringIO
+
+import pytest
+
+import llnl.util.filesystem as fs
+
+import spack.hash_types as ht
+import spack.modules
+import spack.environment as ev
+
+from spack.cmd.env import _env_create
+from spack.spec import Spec
+from spack.main import SpackCommand, SpackCommandError
+from spack.stage import stage_prefix
+
+from spack.util.mock_package import MockPackageMultiRepo
+import spack.util.spack_json as sjson
+from spack.util.path import substitute_path_variables
+
+
+# everything here uses the mock_env_path
+pytestmark = pytest.mark.usefixtures(
+    'mutable_mock_env_path', 'config', 'mutable_mock_repo')
```

Comment:
Is it possible to add a mark that specifies that the following tests require internet access? Do we have such a mark yet?

Reply:
I do not really know anything about this... I do not thing any of this here requires internet per se, but I haven't explicitly checked and I barely know my way around basic pytest let alone the collection of fixtures used here

Reply:
Yeah we should make these tests not use the internet.  I can try to rework this.  I think we can leave @healther to his work at this point 😄